include/miral/miral/display_configuration.h

```diff
@@ -41,6 +45,25 @@ class ConfigurationOption;
 class DisplayConfiguration
 {
 public:
+    /// A class providing access to an arbitrary piece of data from the [DisplayConfiguration].
+    /// This is specifically useful when a user wants to extend the based display configuration
+    /// with some sort of custom payload (e.g. a user may want to extend the layout configuration
+    /// with information describing the position and size of specific applications).
+    /// \remark Since MirAL 5.3
+    class Node
+    {
+    public:
+        explicit Node(YAML::Node const& node);
```

Comment:
1. This is the only use of `YAML::Node` in the API. That smells like an implementation detail being leaked.
2. We don't expect users of the API to create these objects

So can we make the constructor private and make the parameter something we own (e.g. `std::unique_ptr<Self>`)?